server/entity/src/lib.rs

```diff
@@ -38,17 +40,26 @@ pub fn previous_position_velocity_reset(world: &mut World) {
     <(Read<Position>, Write<PreviousPosition>)>::query().par_for_each_mut(
         world.inner_mut(),
         |(pos, mut previous_pos)| {
-            previous_pos.0 = *pos;
+            previous_pos.0.replace(*pos);
         },
     );
     <(Read<Velocity>, Write<PreviousVelocity>)>::query().par_for_each_mut(
         world.inner_mut(),
         |(vel, mut previous_vel)| {
-            previous_vel.0 = vel.0;
+            previous_vel.0.replace(vel.0);
         },
     );
 }
 
+#[fecs::event_handler]
```

Comment:
This event handler will have to be added in the list in `server/src/event_handlers.rs` for it to be registered - right now, as far as I can tell, it will never be invoked.